src/__lib__/CapiComponent.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+import { Component } from 'preact';
+import { connect } from 'unistore/preact';
+
+const getExistingHtml = identifier => {
+    if (typeof document === 'undefined') {
+        return;
+    }
+
+    const node = document.querySelector(`[${identifier}]`);
+
+    return node && node.innerHTML;
+}
+
+const getComponent = (Component, props, html) => (
+    <Component
+        {...props}
+        dangerouslySetInnerHTML={{
+            __html: html,
+        }}
+    />
+);
+
+const htmlStore = {};
+
+export const registeredCapiComponents = new Set();
+
+export const CapiComponent = (MyComponent, contentReference) => {
+    const identifier = `data-content-${contentReference}`;
```

Comment:
should we use some kind of UUID here? what if page has two instances of a `CapiComponent` that display the same info? since this uses `querySelector` we should get just one set of data back, but if the first component done somtehing bespoke, all other instances will reflect the same changes?

Reply:
hmm though how would the client know what the server UUID was? hmmm...